Internals/CLI.js

```diff
@@ -0,0 +1,110 @@
+const { cli } = require("../Configurations/commands");
+const {
+	Error: GABError,
+} = require("./Errors");
+
+class CLI {
+	constructor (sharder, db, traffic) {
+		this.sharder = sharder;
+		this.db = this.database = db;
+		this.traffic = traffic;
+		this.stdin = process.stdin;
+		this.currentlyRunningCommand = null;
+	}
+
+	/**
+	 * Sets up all the stuff we need
+	 * @returns {CLI}
+	 */
+	setup () {
+		if (this.stdin) this.stdin.on("data", this.listener.bind(this));
+		return this;
+	}
+
+	/**
+	 * Disposes (destroys) all the CLI things.
+	 */
+	dispose () {
+		this.stdin.removeListener("data", this.listener.bind(this));
+	}
+
+	/**
+	 * Listens to the incoming events from stdin
+	 * @param {Buffer} data The incoming data from the stdin
+	 */
+	listener (data) {
+		let strData = data.toString();
+		let trimmedData = strData.trim();
+		if (!this.currentlyRunningCommand) {
+			let [command, ...args] = trimmedData.split(" ");
+			command = command.trim();
+			const cmdData = this.getFullCommand(command);
+			if (!cmdData) return;
+			if (cmdData.data.isMultiline) {
+				this.commandArgs = args.join(" ");
+				this.currentlyRunningCommand = {
+					data: cmdData.data,
+					func: cmdData.func,
+				};
+			} else {
+				try {
+					cmdData.func({ cli: this }, cmdData.data, args.join(" "));
+				} catch (err) {
+					winston.error(`An error occurred while running this command:\n${err.stack}`);
+				}
+			}
+		} else {
+			if (this.currentlyRunningCommand.data.isMultiline) {
+				if (trimmedData.endsWith(`\u0007`)) {
+					trimmedData = trimmedData.replace(/\u0007/, "");
```

Comment:
This line seems to be making an eslint error occur, eventhough this shouldn't have to be regex (You don't use the global flag, so this can be a simple string rather than a Regex)

Either turn this into a string, or fix the regex (and, if desired, add the global flag)

Reply:
oh wait i found out one mistake by myself. This regex. Misses $.

Reply:
@GilbertGobbels If disabling the eslint error in 3080e46 shouldn't be done, let me know.